ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+                            f"Did not find field '{anno_unfilled_target_field_name}' for struct '{struct.get_name()}', anno {anno_decl}")
+                else:
+                    # for the parameter, we need the type (the struct that it is definitely referring to)
+                    found_field = False
+
+                    spec_params = spec.get_parameters()
+
+                    for param in spec_params:
+                        if (found_field):
+                            break
+
+                        if param.get_index() == param_index:
+                            # Assumption: param.param_type looks like 'struct <struct_name>' or '<struct_name>'
+                            logout(
+                                f"param_type_struct_name = '{param.get_param_type()}'")
+                            try:
+                                param_type_struct_name = param.get_param_type().split(" ")[
+                                    1]
+                            except IndexError:
+                                param_type_struct_name = param.get_param_type()
+
+                            struct = specifier_manager.get_struct(
+                                param_type_struct_name)
+
+                            for field in struct.get_fields():
+                                if anno_unfilled_target_field_name == field.get_field_name():
+                                    anno_unfilled_target_field_name = field.get_field_index()
+                                    found_field = True
+                                    break
+                            else:
+                                raise ValueError(
+                                    f"Did not find field '{anno_unfilled_target_field_name}' for struct '{struct.get_name()}', anno {anno_decl}")
+
+                    if (not found_field):
+                        raise ValueError(
+                            f"Did not find field '{anno_unfilled_target_field_name}' for anno '{anno_decl}', specifier '{spec.get_name()}'")
+
+                logout(f"UNFILLED_FIELD = {anno_unfilled_target_field_name}")
+                known_target += f".FIELD({anno_unfilled_target_field_name})"
+
+        logout(
+            f"for anno {anno_decl} known target is {known_target} and {spec.get_name()}")
+
+        return AnnotateAttr(anno_type, known_target, anno_methods)
+
+    def raw_ast_to_decl_type(self, ast_line: str, spec_manager: SpecifierManager) -> DeclType:
+        expr = ast_line.strip()
+        first_letter_index = expr.find(next(filter(str.isalpha, expr)))
+        start_decl_index = first_letter_index
+        decl_name = ""
+        cur_decl_char = expr[start_decl_index]
+
+        try:
+            while (cur_decl_char != " "):
+                decl_name += cur_decl_char
+                start_decl_index += 1
+                cur_decl_char = expr[start_decl_index]
+        except IndexError:
+            # might be a null statement (looks like "<<<NULL>>>" in AST)
```

Comment:
This pattern appears at least one other place (`ast_info_generator.py`, around line 40). Can you write a helper function to handle this situation, so that if you do actually fix the `<<<NULL>>>` parsing issue, you only have to do so in one place?

Reply:
As a general rule, copying a TODO statement is an anti-pattern: write a helper function instead so that you only have one copy of the TODO.